kartothek/io/dask/dataframe.py

```diff
@@ -606,3 +607,53 @@ def hash_dataset(
             .apply(_unpack_hash, unpack_meta=ddf._meta, subset=subset, meta="uint64")
             .astype("uint64")
         )
+
+
+def repartition_ddf(
+    ddf: dd.DataFrame,
+    dataset_factory: DatasetFactory,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+    predicates: Optional[PredicatesType] = None,
+) -> dd.DataFrame:
+    """
+    This is an IO  function used to repartition the input ddf using the target_nrows_per_partition input param
+    The metadata of the ddf is collected using Ktk Functions and expected_nrows_per_partition is extracted`.
+    An intermediate parameter repartition_ratio (per_partition) which is target_rows/expected_rows is computed.
+    Final target number of partitions (target_npartitions) are calculated  on the ddf using the above ratio.
+    The input ddf is repartitioned using target_npartitions value.
+
+    Parameters
+    ----------
+    ddf
+        dask.DataFrame
+    dataset_factory
+        dataset factory
+    predicates
+        Optional , but predicates can be applied while collecting_dataset_metadata
+    target_nrows_per_partition
+        target num of rows per partition
+    dataset_metadata_frac
+        This fraction is helps in collecting metadata on a fraction(sample) of the ddf, instead of the whole ddf.
+
+
```

Comment:
I was thinking if we should add `chunksize`as dask allows i.e. the uncompressed size per partition. That is probably more valuable than `target_nrows_per_partition`

Reply:
We should fix about this Parameter as this would be the customer facing one. 
@fjetter , any thoughts here?